index.js

```diff
@@ -33,6 +33,32 @@ exports.register = (server, pluginOptions, next) => {
       }
     };
   });
+  /* will call server.auth.strategy
+   package should be of the form:
+   strategy: {
+    name: 'myStrategyName',
+    mode: true // (can be any valid strategy mode)
+    apiKeys: [
+      [
+        'anAPIKey',
+        {
+          name: 'authenticationName'
+        }
+      ]
+    ]
+   }
+  */
+  if (pluginOptions.strategy) {
+    const options = {
+      apiKeys: {}
+    };
+    pluginOptions.strategy.apiKeys.forEach((apikey) => {
```

Comment:
Right now we pass the api keys for a given strategy as an array of objects like so:

strategy: {
    name: 'myStrategyName',
    mode: true // (can be any valid strategy mode)
    apiKeys: [
      {
        key: 'anAPIKey',
        credentials: {
          name: 'authenticationName'
        }
    ]
   }

  I'd rather apiKeys just be an object, with each key being the api key and the value being the credentials object.  But confi (or someone) doesn't seem to be evaluating the handlebars expressions when they are in the keys of yaml.  If we can get around that obstacle we can have apiKeys just be a simple object.